platform/efm32/inc/configurations.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 Silicon Laboratories, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __CONFIGURATIONS_H__
+#define __CONFIGURATIONS_H__
+
+/*******************************************************************************
+ * Universal configurations
+ ******************************************************************************/
+
+/* Memory boundaries */
+#define FLASH_ORIGIN 0x00000000
+
+/* Memory boundaries */
+#define SRAM_ORIGIN  0x20000000
+#define SRAM_OFFSET  0x0
+
+/* Minimum memory requirements */
+#define FLASH_LENGTH_MIN 0x10000  /* 64k */
+#define SRAM_LENGTH_MIN  0x8000   /* 32k */
+
+/*******************************************************************************
+ * Hardware-specific configurations
+ *
+ * Configurations are named after the parameter values, in this order:
+ *   - CORE
+ *   - PLATFORM
+ ******************************************************************************/
+#if defined(CONFIGURATION_EFM32_M3_P1)
+    /* ARM core selection */
+#   define CORE_CORTEX_M3
```

Comment:
@akselsm I agree with @AlessandroA now - for the time being it's important for us to establish best practice for future ports. The patch should reflect the best case for existing platforms. Once you're requirements change, it's easy to change back. The readers of these future changes will understand better the reasoning.